agent/hcp/telemetry/otel_sink_test.go

```diff
@@ -0,0 +1,282 @@
+		"consul.raft.state.leader": {
+			Name:        "consul.raft.state.leader",
+			Description: "",
+			Unit:        "",
+			Data: metricdata.Sum[float64]{
+				DataPoints: []metricdata.DataPoint[float64]{
+					{
+						Attributes: *attribute.EmptySet(),
+						Value:      float64(float32(23.23)),
+					},
+				},
+			},
+		},
+		"consul.raft.apply": {
+			Name:        "consul.raft.apply",
+			Description: "",
+			Unit:        "",
+			Data: metricdata.Sum[float64]{
+				DataPoints: []metricdata.DataPoint[float64]{
+					{
+						Attributes: attrs,
+						Value:      float64(float32(1.44)),
+					},
+				},
+			},
+		},
+		"consul.raft.leader.lastContact": {
+			Name:        "consul.raft.leader.lastContact",
+			Description: "",
+			Unit:        "",
+			Data: metricdata.Histogram[float64]{
+				DataPoints: []metricdata.HistogramDataPoint[float64]{
+					{
+						Attributes: *attribute.EmptySet(),
+						Count:      1,
+						Sum:        float64(float32(45.32)),
+						Min:        metricdata.NewExtrema(float64(float32(45.32))),
+						Max:        metricdata.NewExtrema(float64(float32(45.32))),
+					},
+				},
+			},
+		},
+		"consul.raft.commitTime": {
+			Name:        "consul.raft.commitTime",
+			Description: "",
+			Unit:        "",
+			Data: metricdata.Histogram[float64]{
+				DataPoints: []metricdata.HistogramDataPoint[float64]{
+					{
+						Attributes: attrs,
+						Count:      1,
+						Sum:        float64(float32(26.34)),
+						Min:        metricdata.NewExtrema(float64(float32(26.34))),
+						Max:        metricdata.NewExtrema(float64(float32(26.34))),
+					},
+				},
+			},
+		},
+	}
+)
+
+func TestNewOTELSink(t *testing.T) {
+	t.Parallel()
+	for name, test := range map[string]struct {
+		wantErr string
+		opts    *OTELSinkOpts
+	}{
+		"failsWithEmptyLogger": {
+			wantErr: "ferror: provide valid context",
+			opts: &OTELSinkOpts{
+				Reader: metric.NewManualReader(),
+			},
+		},
+		"failsWithEmptyReader": {
+			wantErr: "ferror: provide valid reader",
+			opts: &OTELSinkOpts{
+				Reader: nil,
+				Ctx:    context.Background(),
+			},
+		},
+	} {
+		test := test
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			sink, err := NewOTELSink(test.opts)
+			if test.wantErr != "" {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), test.wantErr)
+				return
+			}
+
+			require.NotNil(t, sink)
+		})
+	}
+}
+
+// TestOTELSink performs concurrent metric instrument operations on the sink.
+// When run with go test -race, this test should pass if implementation is concurrency safe.
+func TestOTELSink(t *testing.T) {
+	t.Parallel()
+
+	// Manual reader outputs the aggregated metrics when reader.Collect is called.
+	reader := metric.NewManualReader()
+
+	ctx := context.Background()
+	opts := &OTELSinkOpts{
+		Reader: reader,
+		Ctx:    ctx,
+	}
+
+	sink, err := NewOTELSink(opts)
+	require.NoError(t, err)
+
+	labels := []gometrics.Label{
+		{
+			Name:  "server.id",
+			Value: "test",
+		},
+	}
+
+	wg := &sync.WaitGroup{}
```

Comment:
@chapmanc converted this test to use wait group and perform instrumentation concurrently to test concurrency assumptions.

Reply:
I'll update this test to do something similar to what's done in the gauge store